cache/lru_cache.cc

```diff
@@ -120,7 +120,9 @@ LRUCacheShard::LRUCacheShard(
       strict_capacity_limit_(strict_capacity_limit),
       high_pri_pool_ratio_(high_pri_pool_ratio),
       high_pri_pool_capacity_(0),
-      low_pri_pool_ratio_(low_pri_pool_ratio),
+      low_pri_pool_ratio_(low_pri_pool_ratio < 0.0 || low_pri_pool_ratio > 1.0
```

Comment:
@gangliao I see a bunch of test failures with the latest version; I suspect they're caused by this line right here. Note that the logical OR (`||`) has higher precedence that the ternary (`? :`), so the way it's written, this is equivalent to `low_pri_pool_ratio < 0.0 || (low_pri_pool_ratio > 1.0 ? 1.0 - high_pri_pool_ratio : low_pri_pool_ratio)`, which is a logical value that will end up getting converted to either 0.0 or 1.0. We would want to put the two parts of the range check in parentheses to fix this.

Reply:
Not just here; 

```c++
  if (low_pri_pool_ratio < 0.0 || low_pri_pool_ratio > 1.0) {
    // Invalid low_pri_pool_ratio
    return nullptr;
  }
  if (low_pri_pool_ratio + high_pri_pool_ratio > 1.0) {
    // Invalid high_pri_pool_ratio and low_pri_pool_ratio combination
    return nullptr;
  }
```

Looks like we have to move those checks to another place, otherwise, it already returned nullptr for cache.

Reply:
Please disregard, this should actually work as-is b/c of the precedence rules

Reply:
Or, we don't even need those checks since we change low_pri_pool_ratio automatically. (1 - high_pri_pool_ratio)

Reply:
Do all LRU cache creations eventually hit the code path you mentioned above? If yes, that would be the perfect place to adjust `low_pri_pool_ratio`

Reply:
Okay, for the user interface, yeah, that's true.

Reply:
That is, instead of returning `nullptr` for out-range-values, just set `low_pri_pool_ratio` to `1 - high_pri_pool_ratio`